text/0193-fixing-type-unions.md

```diff
@@ -0,0 +1,581 @@
+---
+feature name: fixing-type-unions
+start date: 2020-07-06
+rfc pr: 194
+related issue: 193
+---
+
+# Summary
+
+_Type unions_ is a feature of the TypeScript language that allows typing values as "one of several candidate types". This is achieved by specifying a
+`|`-delimited list of candidate types (e.g: `Foo | Bar`). While this feature is common in dynamic languages (**TypeScript**, **Python**, ...), it is
```

Comment:
Sorry, but:

- **Rust** unions are unions in the `C` union sense (and hence they require a name). The semantics are very similar, but while in TypeScript you can pass "Foo" or "Bar" to a parameter typed "type FooBar = Foo | Bar", you **must** pass a FooBar value in **Rust**.
- **Swift** enumerations are not type unions, and they must be named. The semantics are similar, and I can agree they are union-like in the same sense that the classes we'd generate per this RFC in Java/C# are union-like.
- **Kotlin** appears to apply ["type decaying"](https://kotlinlang.org/spec/type-system.html#type-decaying) to union types. I found I am not smart enough to understand the description of this process, so I have no idea what it entails 🤷🏻‍♂️.

I honestly don't know enough about the `OCaml` and `Scala` scenarios.

Anyhow - I guess I see how the statement can be perceived as excessively broad, largely because the "type unions" feature is under-qualified (in TypeScript the unions can be anonymous, they imply a value typed as any candidate can be passed through or returned from a union-typed API), and the language groups aren't qualified well enough (as you had noted previously).